compiler/frontend/bs_ast_invariant.ml

```diff
@@ -85,7 +85,9 @@ let emit_external_warnings : iterator =
             ( Nonrecursive,
               [{ptype_kind = Ptype_variant ({pcd_res = Some _} :: _)}] ) ->
           Location.raise_errorf ~loc:str_item.pstr_loc
-            "GADT has to be recursive types, please try `type rec'"
+            "GADTs require recursive type syntax.\n\
+             Please define your type using `type rec` instead of `type`.\n\
```

Comment:
I think we should avoid saying "please" here, we don't do that in other error messages either.